File: Sorting/JavaScript/insertionSortBasic.js
// sort the given array using insertion sort
const arr = [15, 11, 8, 3, 21];

console.log("sorted array: ", arr)

for (let i = 0; i < arr.length - 1; i++) {
  for (let j = i + 1; j > 0; j--) {
    if (arr[j] > arr[j - 1]) {
      break;
    }
    const temp = arr[j];
    arr[j] = arr[j - 1]
    arr[j - 1] = temp;
  }
}

console.log("sorted array: ", arr)

File: Sorting/src/com/vinayak/InsertionSortBasic.java
package com.vinayak;

import java.util.Arrays;

public class InsertionSortBasic {
    public static void main(String[] args) {
        int[] arr = {5, 1, 15, 11, 18};
        System.out.println("unsorted array: " + Arrays.toString(arr));

        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = i + 1; j > 0; j--) {
                if (arr[j] > arr[j - 1]) {
                    break;
                }
                int temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;
            }
        }
        System.out.println("sorted array: " + Arrays.toString(arr));
    }
}
